internal/controller/storageclient_controller.go

```diff
@@ -692,16 +692,33 @@ func (r *storageClientReconcile) reconcileResource(obj client.Object, rawObject
 		}
 		return nil
 	}
-
 	var err error
-	if useReplace {
-		err = utils.CreateOrReplace(r.ctx, r.Client, obj, mutateFunc)
-	} else {
-		_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, obj, mutateFunc)
-	}
+	_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, obj, mutateFunc)
+	if utils.IsForbiddenError(err) {
+		key := client.ObjectKeyFromObject(obj)
+		if err := r.Client.Get(r.ctx, key, obj); client.IgnoreNotFound(err) != nil {
+			return fmt.Errorf("failed to replace %v %v: %v", obj.GetObjectKind().GroupVersionKind(), obj.GetName(), err)
+		}
```

Comment:
There is no reason to try to read it again. It was already read and mutated in the create of update function